src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/AttrHider.java

```diff
@@ -0,0 +1,172 @@
+						return;
+					}
+					final ItemMeta meta = item.getItemMeta();
+					final ItemStack fakeItem = item.clone();
+					if (meta != null) {
+						final ItemMeta fakeMeta = fakeItem.getItemMeta();
+						if (meta instanceof LeatherArmorMeta) {
+							final LeatherArmorMeta leatherMeta = (LeatherArmorMeta) meta;
+							final LeatherArmorMeta fakeLeatherMeta = (LeatherArmorMeta) fakeMeta;
+							fakeLeatherMeta.setColor(leatherMeta.getColor());
+						}
+						if (meta instanceof PotionMeta) {
+							final PotionMeta potionMeta = (PotionMeta) meta;
+							final PotionMeta fakePotionMeta = (PotionMeta) fakeMeta;
+							final PotionData basePotion = potionMeta.getBasePotionData();
+							final PotionData fakePotion = new PotionData(basePotion.getType());
+							fakePotionMeta.setBasePotionData(fakePotion);
+						}
+						if (meta instanceof Damageable) {
+							final Damageable damageable = (Damageable) fakeMeta;
+							damageable.setDamage(0);
+						}
+						if (meta.hasEnchants()) {
+							fakeMeta.addEnchant(Enchantment.DURABILITY, 1, true);
+						}
+						fakeItem.setItemMeta(fakeMeta);
+					}
+					packet.getItemModifier().write(0, fakeItem);
+				}
+			});
+		}
+		if (this.hideEffects) {
+			this.packets.addAdapter(new PacketAdapter(this.plugin, PacketType.Play.Server.ENTITY_EFFECT) {
+				@Override
+				public void onPacketSending(final PacketEvent event) {
+					final PacketContainer packet = event.getPacket();
+					final Player player = event.getPlayer();
+					if (player.hasPermission(BYPASS_PERMISSION)) {
+						return;
+					}
+					final StructureModifier<Integer> ints = packet.getIntegers();
+					if (player.getEntityId() == ints.read(0)) {
+						return;
+					}
+					// set amplifier to 0
+					packet.getBytes().write(1, (byte) 0);
+					// set duration to 0
+					ints.write(1, 0);
+				}
+			});
+		}
+		if (this.hideHealth) {
+			this.packets.addAdapter(new PacketAdapter(this.plugin, PacketType.Play.Server.ENTITY_METADATA) {
+				@Override
+				public void onPacketSending(final PacketEvent event) {
+					final PacketContainer packet = event.getPacket();
+					final Player player = event.getPlayer();
+					if (player.hasPermission(BYPASS_PERMISSION)) {
+						return;
+					}
+					if (packet.getMeta("special").isPresent()) {
+						return;
+					}
+					final Entity entity = packet.getEntityModifier(event).read(0);
+					if (entity == null
+							|| player.getEntityId() == entity.getEntityId()
+							|| !(entity instanceof LivingEntity)
+							|| entity.getPassengers().contains(player)) {
+						return;
+					}
+					for (final WrappedWatchableObject object : packet.getWatchableCollectionModifier().read(0)) {
+						// Read the 8th field as a float as that's the living entity's health
+						// https://wiki.vg/Entity_metadata#Living_Entity
+						if (object.getIndex() == 8) {
+							if ((float) object.getValue() > 0) {
+								object.setValue(1f);
```

Comment:
This is using the broken version of AttrHider which will cause player's health to flash at half a heart

Reply:
Yeah, I've seen the changes to AttrHider, which is why the special meta is included, but I'm completely at as to how that solution you PR'd actually works, because it seems to just cancel that packet and emit an identical one. Why must a new packet be emitted? Why can't the current one be amended? What am I missing here?

Reply:
Because it was broken and this is the only solution I found after hours of work

Reply:
Is there a way to easily replicate the half heart thing?

Reply:
No, it was always a 1 out of 1000, could never reliable reproduce it

Reply:
That begs the question though: if there's no reliable way to reproduce this, how do we know if it was fixed?

Reply:
Hmmm, question, do combat loggers have anything to do with this?